File: src/Api/Features/Modules/Module.cs
using Api.Features.Shared;

namespace Api.Features.Modules;

public class Module : AuditableEntity
{
    public Guid Id { get; set; }
    public required string VariableName { get; set; }
    public required string Label { get; set; }
    public string? Description { get; set; }
    public int VersionNumber { get; set; } = 1;
    public Guid? ParentModuleId { get; set; }
    public Module? ParentModule { get; set; }
    public string? Instructions { get; set; }
    public required string Status { get; set; } = "Active";
    public string? StatusReason { get; set; }
    public bool IsActive { get; set; } = true;
    
    public ICollection<Module> ChildModules { get; set; } = new List<Module>();
    public ICollection<ModuleVersion> Versions { get; set; } = new List<ModuleVersion>();
}

File: src/Designer/src/types/module.ts
export interface Module {
  id: string;
  variableName: string;
  label: string;
  description?: string;
  versionNumber: number;
  parentModuleId?: string;
  instructions?: string;
  status: string;
  statusReason?: string;
  isActive: boolean;
}

export interface ModuleVersion {
  id: string;
  versionNumber: number;
  changeDescription?: string;
  createdOn: string;
  createdBy: string;
}
